src/zcash/Address.hpp

```diff
@@ -23,6 +23,7 @@ class PaymentAddress {
         READWRITE(pk_enc);
     }
 
+    friend inline bool operator==(const PaymentAddress& a, const PaymentAddress& b) { return a.a_pk == b.a_pk && a.pk_enc == b.pk_enc; }
     friend inline bool operator<(const PaymentAddress& a, const PaymentAddress& b) { return a.a_pk < b.a_pk; }
```

Comment:
Yeah, nothing guarantees that adversarially constructed addresses won't have equal a_pk but unequal pk_enc, even though this is vanishingly unlikely for honestly constructed addresses.